pandas/core/dtypes/missing.py

```diff
@@ -452,6 +445,45 @@ def array_equivalent(left, right, strict_nan: bool = False) -> bool:
     return np.array_equal(left, right)
 
 
+def _array_equivalent_float(left, right):
+    return ((left == right) | (np.isnan(left) & np.isnan(right))).all()
```

Comment:
what about use_inf_as_na?

Can this be re-used on L426?  (also the prod(shape) check on L424-425 i think is redundant with earlier shape check)

Reply:
It seems like 1.0.5 doesn't care about that

```python
In [15]: df1 = pd.DataFrame({"A": np.array([np.nan, 1, np.inf])})

In [16]: df2 = pd.DataFrame({"A": np.array([np.nan, 1, np.nan])})

In [17]: with pd.option_context('mode.use_inf_as_na', True):
    ...:     print(df1.equals(df2))
    ...:
    ...:
False
```

Reply:
Makes sense, thanks for checking